File: Chess/src/Tile.js
"use strict"

class Tile {
    constructor(xPos, yPos) {
        this.xPos_ = xPos;
        this.yPos_ = yPos;
        this.cellId_ = xPos + ", " +yPos;
        this.piece_ = null;
        this.cell_ = document.getElementById(this.cellId_ );
    }

    getPiece() {
        this.removePiece()
        return this.piece_;
    }

    removePiece() {
        while (this.cell_.firstChild) {
            this.cell_.removeChild(this.cell_.firstChild);
        }
    }

    setPiece(piece) {
        this.piece_ = piece;
        this.cell_.appendChild(this.piece_.getImageNode());
    }

    getX() {
        return this.xPos_;
    }

    getY() {
        return this.yPos_;
    }

    getCellId(){
        return this.cellId_;
    }
}

File: Chess/src/Tile.java
import javafx.scene.layout.StackPane;
import javafx.scene.shape.Rectangle;

public class Tile extends StackPane {
    private int xPos_;
    private int yPos_;
    private Piece piece_;
    private Rectangle square_;

    Tile(int xPos, int yPos){
        xPos_ = xPos;
        yPos_ = yPos;
        piece_ = null;
    }

    public Piece getPiece(){
        return piece_;
    }
    public void setPiece(Piece piece){
//        remove the current
        try {
            System.out.println(piece_.getName() + " has been killed.");
            this.getChildren().remove(piece_.image_);
        }
        catch (NullPointerException e){
            System.out.println("This spot doesn't have a piece");
        }

        //set the new
        piece_ = piece;
        this.getChildren().add(piece_.image_);
    }
    public int getX(){
        return xPos_;
    }
    public int getY(){
        return yPos_;
    }
}
